src/main/resources/openapi/EducationAndWorkPlanAPI.yml

```diff
@@ -0,0 +1,291 @@
+
+#
+# --------------------------------------------------------------------------------
+#
+components:
+  #
+  # Schema and Enum Definitions
+  # --------------------------------------------------------------------------------
+  schemas:
+    ActionPlanResponse:
+      title: ActionPlanResponse
+      type: object
+      description: A Prisoner's Action Plan containing a list of one or more Goals.
+      properties:
+        prisonNumber:
+          type: string
+          description: The ID of the prisoner
+          example: 'A1234BC'
+        goals:
+          type: array
+          description: A List of at least one or more Goals.
+          items:
+            $ref: '#/components/schemas/GoalResponse'
+      required:
+        - prisonNumber
+        - goals
+    GoalResponse:
+      title: GoalResponse
+      type: object
+      description: An individual Goal that a Prisoner aims to complete as part of their Action Plan. The goal will be active to start with.
+      properties:
+        goalReference:
+          type: string
+          description: The Goal's unique reference
+          example: c88a6c48-97e2-4c04-93b5-98619966447b
+        title:
+          type: string
+          description: A title explaining the aim of the goal.
+          example: Improve communication skills
+        reviewDate:
+          type: string
+          format: date
+          description: An ISO-8601 date representing when the Goal is up for review.
+          example: '2023-12-19'
+        status:
+          $ref: '#/components/schemas/GoalStatus'
+        steps:
+          type: array
+          minItems: 1
+          description: A List of at least one Step.
+          items:
+            $ref: '#/components/schemas/StepResponse'
+        notes:
+          type: string
+          description: Some additional notes related to the Goal.
+          example: Pay close attention to Peter's behaviour.
+        createdBy:
+          type: string
+          description: The name/ID of the person who created the goal. **Note that this is a placeholder for now currently not populated.**
+          example: 'TBD'
+        createdAt:
+          type: string
+          format: date-time
+          description: An ISO-8601 timestamp representing when the Goal was created.
+          example: '2023-06-19T09:39:44Z'
+        updatedBy:
+          type: string
+          description: The name/ID of the person who last updated the goal. **Note that this is a placeholder for now currently not populated.**
+          example: 'TBD'
+        updatedAt:
+          type: string
+          format: date-time
+          description: An ISO-8601 timestamp representing when the Goal was last updated. This will be the same as the created date if it has not yet been updated.
+          example: '2023-06-19T09:39:44Z'
+      required:
+        - goalReference
+        - title
+        - reviewDate
+        - status
+        - createdBy
```

Comment:
I know you've now added the minItems value to the steps array 👍 , but I think we also need steps in the required array as well?
IE. its a mandatory property AND it needs to have at least 1 item in the array
(rather than its an optional property, but if its provided it needs to have at least 1 item in the array)